src/Wolf.cpp

```diff
@@ -0,0 +1,466 @@
+                                      const std::vector<cbmc::TrialMol> &oldMol,
+                                      const std::vector<uint> &molIndexNew,
+                                      const std::vector<uint> &molIndexold,
+                                      bool first_call)
+{
+  return 0.0;
+}
+
+
+//calculate self term after swap move
+double Wolf::SwapSelf(const cbmc::TrialMol& trialMol) const
+{
+  uint box = trialMol.GetBox();
+  if (box >= BOXES_WITH_U_NB)
+    return 0.0;
+
+  GOMC_EVENT_START(1, GomcProfileEvent::SELF_SWAP);
+  MoleculeKind const& thisKind = trialMol.GetKind();
+  uint atomSize = thisKind.NumAtoms();
+  double en_self = 0.0;
+
+  for (uint i = 0; i < atomSize; i++) {
+    en_self -= (thisKind.AtomCharge(i) * thisKind.AtomCharge(i));
+  }    
+
+  GOMC_EVENT_STOP(1, GomcProfileEvent::SELF_SWAP);
+  // M_2_SQRTPI is 2/sqrt(PI), so need to multiply by 0.5 to get sqrt(PI)
+  // Vlugt
+  //self *= ((ff.wolfAlpha[box] * M_2_SQRTPI * 0.5) +  0.5 * ff.wolfFactor1[box]);
+  // we eliminate the alpha/root(pi) using Wolf,mod from Gross et al
+  return (en_self *= -1.0 * ff.wolfFactor1[box] * num::qqFact);
+}
+
+//calculate correction term after swap move
+double Wolf::SwapCorrection(const cbmc::TrialMol& trialMol) const
+{
+  uint box = trialMol.GetBox();
+  if (box >= BOXES_WITH_U_NB)
+    return 0.0;
+
+  GOMC_EVENT_START(1, GomcProfileEvent::CORR_SWAP);
+  double dist, distSq;
+  double correction = 0.0;
+  XYZ virComponents;
+  const MoleculeKind& thisKind = trialMol.GetKind();
+  uint atomSize = thisKind.NumAtoms();
+
+  for (uint i = 0; i < atomSize; i++) {
+    for (uint j = i + 1; j < atomSize; j++) {
+      if(currentAxes.InRcut(distSq, virComponents, trialMol.GetCoords(),
+                         i, j, box)){
+        correction -= thisKind.AtomCharge(i) * thisKind.AtomCharge(j) * wolfFactor1[box];
+      }
+    }
+  }
+  GOMC_EVENT_STOP(1, GomcProfileEvent::CORR_SWAP);
+  return num::qqFact * correction;
+}
+//calculate correction term after swap move
+double Wolf::SwapCorrection(const cbmc::TrialMol& trialMol,
+                               const uint molIndex) const
+{
+  uint box = trialMol.GetBox();
+  if (box >= BOXES_WITH_U_NB)
+    return 0.0;
+
+  GOMC_EVENT_START(1, GomcProfileEvent::CORR_SWAP);
+  double dist, distSq;
+  double correction = 0.0;
+  XYZ virComponents;
+  const MoleculeKind& thisKind = trialMol.GetKind();
+  uint atomSize = thisKind.NumAtoms();
+  uint start = mols.MolStart(molIndex);
+  double lambdaCoef = GetLambdaCoef(molIndex, box);
+
+  for (uint i = 0; i < atomSize; i++) {
+    if(particleHasNoCharge[start + i]) {
+      continue;
+    }
+    for (uint j = i + 1; j < atomSize; j++) {
+      if(currentAxes.InRcut(distSq, virComponents, trialMol.GetCoords(),
+                         i, j, box)){
+        // We eliminate the dampened rij term by choosing psi = 1
+        correction -= thisKind.AtomCharge(i) * thisKind.AtomCharge(j) *
+                      wolfFactor1[box];
```

Comment:
The rcut that boxDim uses to check the distance is max(rcut, rcutcoulomb). You need to additionally check for rcutcoulomb, otherwise if user set a cutoff that `rcut > rcutcoulomb` the results will be wrong.